internal/events/eventhandlers/payment_to_submitter_event_handler.go

```diff
@@ -0,0 +1,82 @@
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/events"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/events/schemas"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/services"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/utils"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/router"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/tenant"
+)
+
+type PaymentToSubmitterEventHandlerOptions struct {
+	DBConnectionPool    db.DBConnectionPool
+	TSSDBConnectionPool db.DBConnectionPool
+	CrashTrackerClient  crashtracker.CrashTrackerClient
+}
+
+type PaymentToSubmitterEventHandler struct {
+	tenantManager      tenant.ManagerInterface
+	crashTrackerClient crashtracker.CrashTrackerClient
+	service            services.PaymentToSubmitterServiceInterface
+}
+
+var _ events.EventHandler = new(PaymentToSubmitterEventHandler)
+
+func NewPaymentToSubmitterEventHandler(options PaymentToSubmitterEventHandlerOptions) *PaymentToSubmitterEventHandler {
+	tm := tenant.NewManager(tenant.WithDatabase(options.DBConnectionPool))
+	tr := router.NewMultiTenantDataSourceRouter(tm)
+	mtnDBConnectionPool, err := db.NewConnectionPoolWithRouter(tr)
+	if err != nil {
+		log.Fatalf("error getting tenant DB Connection Pool: %s", err.Error())
+	}
+
+	models, err := data.NewModels(mtnDBConnectionPool)
+	if err != nil {
+		log.Fatalf("error getting models: %s", err.Error())
+	}
+
+	s := services.NewPaymentToSubmitterService(models, options.TSSDBConnectionPool)
+
+	return &PaymentToSubmitterEventHandler{
+		tenantManager:      tm,
+		service:            s,
+		crashTrackerClient: options.CrashTrackerClient,
+	}
+}
+
+func (h *PaymentToSubmitterEventHandler) Name() string {
+	return "PaymentToSubmitterEventHandler"
+}
+
+func (h *PaymentToSubmitterEventHandler) CanHandleMessage(ctx context.Context, message *events.Message) bool {
+	return message.Topic == events.PaymentCompletedTopic
+}
+
+func (h *PaymentToSubmitterEventHandler) Handle(ctx context.Context, message *events.Message) {
+	paymentsReadyToPay, err := utils.ConvertType[any, schemas.EventPaymentsReadyToPayData](message.Data)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, fmt.Sprintf("[PaymentToSubmitterEventHandler] could not convert data to %T: %v", schemas.EventPaymentsReadyToPayData{}, message.Data))
+		return
+	}
+
+	t, err := h.tenantManager.GetTenantByID(ctx, message.TenantID)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, "[PaymentToSubmitterEventHandler] error getting tenant by id")
+		return
+	}
+
+	ctx = tenant.SaveTenantInContext(ctx, t)
+
+	if err := h.service.SendPaymentsReadyToPay(ctx, &paymentsReadyToPay); err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, fmt.Sprintf("[PaymentToSubmitterEventHandler] send payments ready to pay: %s", paymentsReadyToPay.Payments))
+		return
+	}
+}
```

Comment:
💡 Rather than hardcoding the class name on these errors, why don't you use string templates and inject the value of `s.Name()`?